packages/peregrine/src/hooks/useCarousel.js

```diff
@@ -0,0 +1,28 @@
+import { useState, useCallback } from 'react';
+
+export const useCarousel = carouselSize => {
+    const [activeItemIndex, setActiveItemIndex] = useState(0);
+    const [currentImageLoaded, setCurrentImageLoaded] = useState(false);
+
+    const handleImageLoad = () => setCurrentImageLoaded(true);
+
+    const handlePrevious = useCallback(() =>
+        activeItemIndex > 0
+            ? setActiveItemIndex(activeItemIndex - 1)
+            : setActiveItemIndex(carouselSize - 1)
+    );
+
+    const handleNext = useCallback(() =>
+        setActiveItemIndex((activeItemIndex + 1) % carouselSize)
```

Comment:
```suggestion
        setActiveItemIndex((activeItemIndex + 1) % carouselSize), [activeItemIndex]
```
Please refer to above explanation